pkg/cloud/provider.go

```diff
@@ -212,6 +213,39 @@ type ServiceAccountStatus struct {
 	Checks []*ServiceAccountCheck `json:"checks"`
 }
 
+// ServiceAccountChecks is a thread safe map for holding ServiceAccountCheck objects
+type ServiceAccountChecks struct {
+	serviceAccountChecks map[string]*ServiceAccountCheck
+	lock                 *sync.RWMutex
```

Comment:
For what it's worth, I have used `sync.Mutex` pointers exclusively in our code base for a long time, and I actually just had to admit recently that I was wrong to @michaelmdresser as much as it pained me to do so 😄 . Using a pointer here isn't the end of the world, but the problematic case is on a dereferenced copy (you'd get a copy of the resource and a copy of the pointer to the Mutex). This sounds really bad, and it is, but dereferencing pointer types in general will typically not end well (especially using any sort of complex type). For instance, even if you were to change the mutex here, if someone deref copied a ServiceAccountChecks instance, the map would not be shared between two instances as well. 

So while I have generally been a preference based advocate for using `new(sync.Mutex)` in the past, I think the better/safer choice here is to use either `lock sync.RWMutex` or embed `sync.RWMutex`. I will follow suit in my future code writing as well to ensure we don't end up with any nasty deref copied shared locks :(

Reply:
thanks for letting me know, I think the embedded makes the most sense here.